endpoint.php

```diff
@@ -254,14 +254,52 @@ function invalidRequest(): void
         exit();
     }
     $revoke = filter_input(INPUT_POST, 'action', FILTER_VALIDATE_REGEXP, ['options' => ['regexp' => '@^revoke$@']]);
+    $token = filter_input(INPUT_POST, 'token', FILTER_VALIDATE_REGEXP, ['options' => ['regexp' => '@^[0-9a-f]+_[0-9a-f]+$@']]);
+    // check if is POST+revoke request
     if (is_string($revoke)) {
-        $token = filter_input(INPUT_POST, 'token', FILTER_VALIDATE_REGEXP, ['options' => ['regexp' => '@^[0-9a-f]+_[0-9a-f]+$@']]);
         if (is_string($token)) {
             revokeToken($token);
         }
         header('HTTP/1.1 200 OK');
         exit();
     }
+    // check if is POST+introspection request
+    if (is_string($token)) {
+        $tokenInfo = retrieveToken($token);
+        if ($tokenInfo === null || $tokenInfo['active'] === '0') {
+            header('HTTP/1.1 200 OK');
+            header('Content-Type: application/json;charset=UTF-8');
+            exit(json_encode([
+                'active' => false,
+            ]));
+        }
+        // Authorize resource server as per specification (see https://indieauth.spec.indieweb.org/#access-token-verification-response-p-1)
+        // For us this means we are expecting the Basic user to be the client ID of the consumer.
+        // With basic, everything after the first colon (:) is considered the password.
+        // Since we are working with URIs to identify, we need to handle
+        // the pattern scheme://domain/path:password
+        // To do this, we assume (and enforce) that the password is a single underscore (_).
+        $basicAuth = $_SERVER['PHP_AUTH_USER'] . ':' . $_SERVER['PHP_AUTH_PW'];
+        $storedAuth = $tokenInfo['auth_client_id'] . ':_';
+        $storedAuthEncoded = urlencode($tokenInfo['auth_client_id']) . ':_';
+        if ($storedAuth !== $basicAuth && $storedAuthEncoded !== $basicAuth) {
+            header('WWW-Authenticate: Basic');
+            header('HTTP/1.0 401 Unauthorized');
+            exit('Unauthorized');
+        }
+        header('HTTP/1.1 200 OK');
+        header('Content-Type: application/json;charset=UTF-8');
+        exit(json_encode([
+            'token_type' => 'Bearer',
+            'me' => $tokenInfo['auth_me'],
```

Comment:
[mod_oauth2](https://github.com/OpenIDC/mod_oauth2) requires the `sub` field. My guess is that it uses this field to set the user field in Apache HTTPd (passed to applications and also logged).
```suggestion
            'me' => $tokenInfo['auth_me'],
            'sub' => $tokenInfo['auth_me'],
```

Reply:
I rhink that is a fine reason to duplicate `me` to `sub` 👍